src/common/event-emitter/contextual-ee.test.js

```diff
@@ -0,0 +1,320 @@
+import { faker } from '@faker-js/faker'
+
+let mockNREUM
+let runtime
+
+beforeEach(() => {
+  mockNREUM = {}
+  runtime = {}
+
+  jest.doMock('../window/nreum', () => ({
+    __esModule: true,
+    gosNREUM: jest.fn(() => mockNREUM)
+  }))
+
+  jest.doMock('../config/config', () => ({
+    __esModule: true,
+    getRuntime: jest.fn(() => runtime)
+  }))
+})
+
+afterEach(() => {
+  jest.resetModules()
+  jest.resetAllMocks()
+})
+
+describe('global event-emitter', () => {
+  test('it returns the event-emitter defined on window.NREUM', async () => {
+    const mockEE = {}
+    mockNREUM.ee = mockEE
+
+    const { ee } = await import('./contextual-ee')
+
+    expect(ee).toEqual(mockEE)
+  })
+
+  test('it sets the global event-emitter on window.NREUM', async () => {
+    const { ee } = await import('./contextual-ee')
+
+    expect(ee).toEqual(mockNREUM.ee)
+  })
+})
+
+describe('scoping event-emitter', () => {
+  test('it creates a new child event-emitter', async () => {
+    const { ee } = await import('./contextual-ee')
+
+    const childName = faker.datatype.uuid()
+    const result = ee.get(childName)
+
+    expect(result).not.toEqual(mockNREUM.ee)
+    expect(result).toEqual(ee.get(childName)) // Should always return the same event-emitter
+    expect(result.debugId).toEqual(childName)
+  })
+
+  test('it always returns the same child event-emitter once created for the given name', async () => {
+    const { ee } = await import('./contextual-ee')
+
+    const childName = faker.datatype.uuid()
+    const result = ee.get(childName)
+
+    expect(result).toEqual(ee.get(childName))
```

Comment:
This test seems like its testing the same thing as the test on `44`

Reply:
No, this test is calling `ee.get` again and comparing the object to the original result. We are testing that the child EE is only created once.

Reply:
Lines 48 and 51 test the same thing that lines 59 and 61 do.

The entire test from 55 to 62 (lets call it test B) is exactly the same as lines 43 to 53 (test A), except that test A just has two other cases it also checks.

This means that test B is a subset of test A, making test B redundant.  Obviously this isn't a big deal, but it's unnecessary because it doesn't tell us anything new from test A.

Reply:
Ah yea you are correct.